packages/manager/modules/pci/src/projects/project/training/job.service.js

```diff
@@ -25,6 +26,31 @@ export default class PciProjectTrainingJobService {
       // remove shutdown before submitting as it is forbidden server side and irrelevant when resubmitting
       toSubmit = (({ shutdown, ...j }) => j)(jobSpec);
     }
+    if (jobSpec.volumes) {
+      // API does not accept volumes' internal property. We remove it before sending the request.
+      toSubmit.volumes = jobSpec.volumes.flatMap((volume) => {
+        let toRemove = false;
+        const internalKeys = [];
+        let filteredVolume = volume;
+        Object.keys(volume).forEach((key) => {
+          if (volume[key] && volume[key].internal !== undefined) {
+            if (volume[key].internal === true) {
+              toRemove = true;
```

Comment:
```suggestion
              return [];
```

Reply:
Returning `[]` in the forEach loop does not work here, as the forEach returns `void` anyway. So without the `toRemove` flag, we will still have the volume being added to the volumes to submit instead of filtering it.